coffee_maker_navigator_2/lib/di/di/injector.dart

```diff
@@ -1,34 +1,79 @@
-import 'package:flutter/material.dart';
-import 'container_manager.dart';
+import 'package:coffee_maker_navigator_2/di/di/dependency_container_manager.dart';
+import 'package:coffee_maker_navigator_2/di/di/dependency_container_subscriber.dart';
+import 'package:flutter/widgets.dart';
 
+/// A widget that provides access to the `DependencyContainerManager` throughout the widget tree.
+///
+/// The `Injector` widget is an `InheritedWidget` that holds a reference to the
+/// `DependencyContainerManager`. It ensures that the DI system is accessible
+/// from anywhere within the widget tree, facilitating the resolution and
+/// management of dependencies.
+///
+/// This widget is used to inject the `DependencyContainerManager` into the widget
+/// tree, making it possible to obtain and manage dependency containers from any
+/// descendant widget.
 class Injector extends InheritedWidget {
-  final ContainerManager containerManager;
+  /// The `DependencyContainerManager` that manages the lifecycle of dependency containers.
+  final DependencyContainerManager containerManager;
 
+  /// Creates an `Injector` widget.
+  ///
+  /// [key]: An optional key for the widget.
+  /// [child]: The child widget which will have access to the DI system.
+  /// [containerManager]: The `DependencyContainerManager` instance to be provided.
   const Injector({
     Key? key,
     required Widget child,
     required this.containerManager,
   }) : super(key: key, child: child);
 
-  static ContainerManager of(BuildContext context) {
+  /// Retrieves the `DependencyContainerManager` from the given `BuildContext`.
+  ///
+  /// This method allows any descendant widget to access the `DependencyContainerManager`
+  /// by calling `Injector.of(context)`.
+  ///
+  /// [context]: The `BuildContext` from which to retrieve the `DependencyContainerManager`.
+  ///
+  /// Returns the `DependencyContainerManager` instance provided by the nearest `Injector` widget.
+  static DependencyContainerManager of(BuildContext context) {
```

Comment:
And from here let's not provide access to manager directly, and just provide this widget as a point of interaction

Reply:
done 15d3f6f564106b90078699237aa9ed28e1f778b7